Content.Server/GameObjects/Components/Doors/ServerDoorComponent.cs

```diff
@@ -463,67 +517,93 @@ public virtual void OnUpdate(float frameTime)
 
             if (AutoClose)
             {
-                OpenTimeCounter += frameTime;
+                _openTimeCounter += frameTime;
             }
 
-            if (OpenTimeCounter > CloseSpeed)
+            var realCloseTime = AutoCloseDelay;
+
+            if(Owner.TryGetComponent(out IDoorCheck? doorCheck))
             {
-                if (!CanClose() || !Close())
+                realCloseTime = doorCheck.GetCloseSpeed() ?? realCloseTime;
+            }
+
+            if (_openTimeCounter > realCloseTime)
+            {
+                if (CanCloseGeneric())
+                {
+                    // Close() resets OpenTimeCounter to 0, so it's fine.
+                    Close();
+                }
+                else
                 {
                     // Try again in 2 seconds if it's jammed or something.
-                    OpenTimeCounter -= 2;
+                    _openTimeCounter -= 2;
                 }
             }
         }
 
-        public enum DoorState
+        public async Task<bool> InteractUsing(InteractUsingEventArgs eventArgs)
         {
-            Closed,
-            Open,
-            Closing,
-            Opening,
-        }
-
-        public virtual async Task<bool> InteractUsing(InteractUsingEventArgs eventArgs)
-        {
-            if (!_canWeldShut)
+            if(!eventArgs.Using.TryGetComponent(out ToolComponent? tool))
             {
-                _beingWelded = false;
                 return false;
             }
 
-            if (!eventArgs.Using.TryGetComponent(out WelderComponent? tool) || !tool.WelderLit)
+            // for prying doors
+            if (tool.HasQuality(ToolQuality.Prying) && !IsWeldedShut)
             {
-                _beingWelded = false;
-                return false;
-            }
+                var successfulPry = false;
 
-            if (_beingWelded)
-                return false;
+                if (Owner.TryGetComponent(out IDoorCheck? doorCheck))
+                {
+                    doorCheck.OnStartPry(eventArgs);
+                    successfulPry = await tool.UseTool(eventArgs.User, Owner,
+                        doorCheck.GetPryTime() ?? 0.2f, ToolQuality.Prying, () => doorCheck.CanPryCheck(eventArgs));
+                }
+                else
+                {
+                    successfulPry = await tool.UseTool(eventArgs.User, Owner, 0.2f, ToolQuality.Prying);
+                }
```

Comment:
0.2 is way too fast considering the check's already finished by the time the do_after arrives to the client. I'd say minimum 0.5